ui/pages/password-reset/index.js

```diff
@@ -0,0 +1,185 @@
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+import { useSWRConfig } from 'swr'
+
+import LoginLayout from '../../components/layouts/login'
+
+export default function PasswordReset() {
+  const { mutate } = useSWRConfig()
+  const router = useRouter()
+  const { token } = router.query
+
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState('')
+  const [submitted, setSubmitted] = useState(false)
+
+  async function onSubmitSetPassword(e) {
+    e.preventDefault()
+
+    try {
+      const res = await fetch('/api/password-reset', {
+        method: 'post',
+        body: JSON.stringify({
+          token,
+          password,
+        }),
+      })
+
+      if (!res.ok) {
+        throw await res.json()
+      }
+
+      await res.json()
+
+      await mutate('/api/users/self')
+      router.replace('/')
+    } catch (e) {
+      console.error(e)
+    }
+
+    return false
+  }
+
+  async function onSubmit(e) {
+    e.preventDefault()
+
+    try {
+      const res = await fetch('/api/password-reset-request', {
+        method: 'post',
+        body: JSON.stringify({
+          email,
+        }),
+      })
+
+      if (!res.ok) {
+        throw await res.json()
+      }
+
+      await res.json()
+      setSubmitted(true)
+    } catch (e) {
+      console.error(e)
+    }
+
+    return false
+  }
+
+  return (
+    <>
+      {token ? (
```

Comment:
overall i think this looks good! to improve this a bit is maybe we can refactor this a bit because I notice many duplicate code. 

so what we can do is to make line 71 to line 123 to a component - `password-reset-form` and pass in props like `{title, value, onSubmit, label, placeholder, onChange, buttonTitle, error }`
And then you can reuse the same component to replace line 71 to 123 and line 129 to 176

Reply:
Hmm. these forms aren't the same. Not sure where the reuse is.

Reply:
oh?O.o I thought they are similar that they both have a form with a label, input and a button.